language/move-lang/src/to_bytecode/translate.rs

```diff
@@ -0,0 +1,1022 @@
+        }
+
+        E::BinopExp(el, op, er) => {
+            exp(context, function_frame, code, el)?;
+            exp(context, function_frame, code, er)?;
+
+            function_frame.popn(2);
+            binary_op(code, op);
+            function_frame.push(eloc)?;
+        }
+
+        E::Pack(s, tys, field_args) => {
+            let num_args = field_args.len();
+            for (_, _, earg) in field_args {
+                exp_(context, function_frame, code, earg)?;
+            }
+            let def_idx = context.struct_definition_index(&s);
+            let local_sig = F::LocalsSignature(base_types(context, tys)?);
+            let local_idx = context.locals_signature_index(eloc, local_sig)?;
+
+            function_frame.popn(num_args);
+            code.push(B::Pack(def_idx, local_idx));
+            function_frame.push(eloc)?;
+        }
+
+        E::ExpList(items) => {
+            for item in items {
+                let ei = match item {
+                    G::ExpListItem::Single(ei, _) | G::ExpListItem::Splat(_, ei, _) => ei,
+                };
+                exp_(context, function_frame, code, ei)?;
+            }
+        }
+
+        E::Borrow(mut_, el, field) => {
+            let sdef_idx = struct_definition_index(context, &el.ty);
+            exp(context, function_frame, code, el)?;
+            let f_idx = context.field(sdef_idx, field);
+
+            function_frame.pop();
+            code.push(if mut_ {
+                B::MutBorrowField(f_idx)
+            } else {
+                B::ImmBorrowField(f_idx)
+            });
+            function_frame.push(eloc)?;
+        }
+
+        E::BorrowLocal(mut_, v) => {
+            let l_idx = function_frame.local(&v);
+            code.push(if mut_ {
+                B::MutBorrowLoc(l_idx)
+            } else {
+                B::ImmBorrowLoc(l_idx)
+            });
+            function_frame.push(eloc)?;
+        }
+    }
+    Ok(())
+}
+
+fn call(
+    context: &mut Context,
+    code: &mut Code,
+    m: ModuleIdent,
+    f: FunctionName,
+    tys: Vec<G::BaseType>,
+) -> Result<()> {
+    use F::Bytecode as B;
+    match (&m.0.value.address, m.0.value.name.value(), f.value()) {
+        (&Address::LIBRA_CORE, "Transaction", "gas_unit_price") => code.push(B::GetTxnGasUnitPrice),
+        (&Address::LIBRA_CORE, "Transaction", "max_gas_units") => code.push(B::GetTxnMaxGasUnits),
+        (&Address::LIBRA_CORE, "Transaction", "gas_remaining") => code.push(B::GetGasRemaining),
+        (&Address::LIBRA_CORE, "Transaction", "sender") => code.push(B::GetTxnSenderAddress),
+        (&Address::LIBRA_CORE, "Transaction", "create_account") => code.push(B::CreateAccount),
+        (&Address::LIBRA_CORE, "Transaction", "sequence_number") => {
+            code.push(B::GetTxnSequenceNumber)
+        }
+        (&Address::LIBRA_CORE, "Transaction", "public_key") => code.push(B::GetTxnPublicKey),
+        (&Address::LIBRA_CORE, "Transaction", "assert") => {
```

Comment:
What about pulling these string constants out in to a separate file under `shared`? This way they can also be shared by the parser, and there's only one source of truth for these names?

Reply:
They aren't used by the parser